applications/GeoMechanicsApplication/custom_utilities/math_utilities.h

```diff
@@ -25,6 +27,16 @@ class KRATOS_API(GEO_MECHANICS_APPLICATION) GeoMechanicsMathUtilities
 public:
     [[nodiscard]] static std::vector<double> CalculateDeterminants(const std::vector<Matrix>& rMatrices);
 
+    template <typename VectorType>
+    [[nodiscard]] static VectorType Normalized(const VectorType& rVector)
+    {
+        KRATOS_ERROR_IF(std::none_of(rVector.begin(), rVector.end(), [](auto Component) {
+            return std::abs(Component) > 0.0;
+        })) << "A zero vector cannot be normalized\n";
```

Comment:
This tries to avoid squares and squareroots. In the past I have seen trouble arising from this. The square of a small number may be something numerically un-representable small. Therefore I would julst check the result of norm_2 before the division. Also in such a way that we don't divide by numbers that then cause numerically un-representable numbers.

Should the implementation be here in the header file?

Reply:
I moved the check to just the norm_2, but we still need to be a discussion on when is a number numerically un-representable and which tolerance do we then use.

To answer your second question: it needs to be in the header, since it's a templated function (we could input e.g. a Vector, an array_1d, just as long as we can perform norm_2 on it and std::abs on the components and compare it to 0,0)

Reply:
As discussed, we will start with this check without tolerance for this PR and think more on checking e.g. the jacobian of the element at an earlier stage